### PR TITLE
Different Bugs and small issues collector

### LINKED ISSUE #33: Remove projectId from queries for embeddings and zero-shot recommendations on for downloaded models
Currently the requests for the embeddings and zero shot recommendation require the projectId on the downloaded models page. This can be solved by setting the projectId as optional parameter and in this case all models should be returned.

### LINKED ISSUE #45: Remove projectId from url on downloaded models.
Currently the requests for the embeddings and zero shot recommendations require the projectId on the downloaded models page. This can be removed from the url and from the requests. With that the downloaded models can be accessed globally not as part of the projects.

### LINKED ISSUE #63: Make deletion icons red on settings page
I think the icons for deletion are not looking threatening enough with their gray stroke:
image

Would love to have it in the dangerously looking red that we have in the data browser:
image

### LINKED ISSUE #131: [UX] - Quick delete for admins in projects overview
For more speed while testing it would be great to have a functionality to quickly delete projects out of the projects overview. Maybe you could turn a admin perspective on and off in which you can delete projects. ATTENTION: only for admins and not for users because it would be to dangerous if someone deletes projects in an accident

### LINKED ISSUE #139: [UX] - Confidence chart scale
**What would you improve?**
I'm not too sure about this one but I wanted to discuss it here. The confidence chart is currently maxed on the y-axis depending on the maximum confidence. If you have e.g. 80% as your maximum confidence, it will be capped at that value.

image

It is actually showing a percentage, such that 100% could also be a valid choice. What I'm fearing if we don't do so, is that e.g. a maximum value at 30% confidence will look "good".

### LINKED ISSUE #164: Provide more information on the integration of lookup lists into exec envs
**Is your feature request related to a problem? Please describe.**
Lookup list can be used in execution environments by using the import statement (form knowledge import ...). This functionality is very helpful but hardly explained in the application. So, if a user doesn't know how to import or forgets, it is hard for him to find the needed information.

**Describe the solution you'd like**
On the look up list page, the variable name can be copied on click. Instead, it would be better to copy the whole import statement on click.
On the attribute calculation or labeling function page, a hint on how to import lookup lists could be placed, or also a list of the available look-up lists (as for attributes).

**Additional context**
test findings v1.5.0

### LINKED ISSUE #167: [BUG] - Label distribution chart displays incorrect numbers
**Describe the bug**
Label distribution chart displays numbers that are incorrect. (Possibly just multiplied by 100)

**To Reproduce**
1. Go to the overview page of your project
2. Hover over the label distribution bars
3. See error

**Expected behavior**
The numbers should be correct.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: chrome
 - Version: 1.5.0

**Additional context**
-

### LINKED ISSUE #168: [UX] - Adding comments in the labeling view will re-label my data
**What would you improve?**
When I input my comment in the labeling view, all the keyboard shortcuts get triggered. That means I will accidentally re-label the data and switch between records when writing a comment. I would expect the keyboard shortcuts to be disabled when writing a comment.

Real-world example:
comment_issue

### LINKED ISSUE #170: [BUG] - Blacklisted terms are still part of the lookup list in functions
**Describe the bug**
When using lookup lists in functions, there is the option to "blacklist" terms. The UI explains blacklisting as follows: "Not part of the lookup list in a function and [...]". When I blacklisted the term "season" in my "sports" lookup list, I was still able to find that term in my lookup-labeling-function, though.

**To Reproduce**
1. Create a lookup list, e.g. called "sports"
2. Add a blacklisted term to it, e.g. "season"
3. Write a labeling function that imports and prints out the contents of the lookup list
4. See the error: the term is in the list printed out

**Expected behavior**
Blacklisted terms should not be part of the list in a function, as indicated by the UI explanation.

**Screenshots**
lookup list bug

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version v1.5.0

**Additional context**

### LINKED ISSUE #171: [UX] - Crowd Heuristic Settings should be fixed with the first label that is created
# UX Issue description
In the current Crowd Heuristic, you can choose a combination of labeling task, annotator, and data slice. However, this combination is not fixed upon selection. That results in the user being able to change this combination even though there is labeled data already. If you now
1. let person A label half of slice A
2. switch the heuristic to have person B label half of slice B

then you cannot separate the results in the statistics making them much less useful. It is also rather confusing to work with.

Screenshots of me changing the combination:
image
image

In the second screenshot, you are presented with an incomprehensible state. It is "finished" but the progress bar is fully empty. There is not a single record labeled on that slice but the statistics show the coverage from the slice before. 

# Suggestion
Fix the combination of labeling task, annotator, and data slice. This can be done as soon as the first label was set because that is basically the start of the statistics being calculated.

### PR DESCRIPTION
**Pull request by JWittmeyer**
Issues
- closes https://github.com/code-kern-ai/refinery/issues/139
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery/issues/168
  - tested by Felix 
- closes https://github.com/code-kern-ai/refinery/issues/164
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery/issues/131
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery/issues/171
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery/issues/172
  - tested by Felix (see remark in comments)
- closes https://github.com/code-kern-ai/refinery-ui/issues/63
  - tested by Felix (see remark in comments)
- rename labels (part of) https://github.com/code-kern-ai/refinery/issues/37
  - tested by Felix (see remark in comments)
- closes https://github.com/code-kern-ai/refinery/issues/175
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery-gateway/issues/33
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery-ui/issues/45
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery/issues/178
  - tested by Felix
- closes https://github.com/code-kern-ai/refinery/issues/177
  - tested by Felix
 
Issues Bugs:
- Fixes https://github.com/code-kern-ai/refinery/issues/167
  - tested by Felix
- Fixes https://github.com/code-kern-ai/refinery/issues/170
  - tested by Felix

PRs:
- https://github.com/code-kern-ai/refinery-ui/pull/84
- https://github.com/code-kern-ai/refinery-gateway/pull/81
- https://github.com/code-kern-ai/refinery-submodule-model/pull/24

### CODE REVIEW

**Comment by FelixKirsch** [comment]
## Review Felix

tested
- https://github.com/code-kern-ai/refinery/issues/139

File: src/app/projects/components/create-new-attribute/create-new-attribute.component.html [hunk 1]
```diff
@@ -73,7 +73,16 @@
                     <ng-template #noAttributes>
                         <div class="text-sm font-normal text-gray-500">No usable attributes.</div>
                     </ng-template>
-
+                </div>
+                <div class="flex flex-row items-center" *ngIf="knowledgeBases">
+                    <div class="text-sm leading-5 font-medium text-gray-700 inline-block mr-2">Knowledge Bases</div>
```
Comment (FelixKirsch) [inline src/app/projects/components/create-new-attribute/create-new-attribute.component.html]: I would change this to Lookup Lists to be consistent in the application.

**Comment by FelixKirsch** [comment]
## Further findings

- [x] fixed
- Dropdown 'Select Annotator' on the crowd labeling page only shows 'User' as option and not the correct user email address
Screenshot 2022-11-15 at 14 17 57

- [x] can't reproduce -> discussed
- When I move as an annotator from the project overview page inside a project, I receive an error in the console and cannot select any task. This issue only arises if a task for the annotator in the project exists
Screenshot 2022-11-15 at 14 22 42

- [x] fixed
- The labeling task names are not centered below the labeling task column name
Screenshot 2022-11-16 at 09 55 30

**Comment by FelixKirsch** [comment]
Regarding https://github.com/code-kern-ai/refinery/issues/172
The problem described in the issue is solved. When a labeling function has been "Run on 10" the log is not replaced with the last "Run" log after changing some code. But if I now use "Run on 10" and use "Run" afterward the log of "Run on 10" is still displayed and I am not able to see the log on "Run".
I would expect here that the log of the last labeling function run is displayed, so "Run on 10" or "Run" depending on which has been run latest.

- [x] Fixed

**Comment by FelixKirsch** [comment]
Regarding https://github.com/code-kern-ai/refinery-ui/issues/63
The delete icons are not aligned horizontally.
Screenshot 2022-11-16 at 09 50 37

_____________________________
pretty much impossible to fix

**Comment by FelixKirsch** [comment]
On setting a new label name, the shortcut for the label is also reset while typing.
Screenshot 2022-11-16 at 10 00 24
Screenshot 2022-11-16 at 10 00 29

- [x] fixed

**Comment by FelixKirsch** [comment]
After renaming a label, the old name is still displayed on the settings page. It changes to the new name when the page is refreshed.

______________________
couldn't reproduce
- [x] discussed

File: src/app/projects/components/project-settings/helper/label-helper.ts [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+        this.renameLabelData.canCheck = this.isValidNewName(input.value);
+        this.renameLabelData.newLabelName = input.value;
+    }
+    private isValidNewName(name: string): boolean {
+        if (!name) return false;
+        if (name == this.currentLabel.label.name) return false;
+        if (name.trim() == '') return false;
+        const found = this.labelMap.get(this.currentLabel.taskId).find(((x: any) => x.name == name));
+        if (found) return false;
```
Comment (FelixKirsch) [inline src/app/projects/components/project-settings/helper/label-helper.ts:146]: The function isLabelNameUnique could be used here.

File: src/app/util/route-manager.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    //     // if (!RouteManager.lastUrls || RouteManager.lastUrls.length == 0) return null;
+    //     // return RouteManager.lastUrls[RouteManager.lastUrls.length - 1];
+    //     return RouteManager.previousUrl;
+    // }
+
+    public static moveBack() {
+        // this.router.navigate(['config'])
+
+        // this.router.navigate(['projects', notification.projectId, notification.page]);
+        // this.router.navigate(["../../settings"], { relativeTo: this.activatedRoute });
```
Comment (FelixKirsch) [inline src/app/util/route-manager.ts:70]: are the comments needed or can they be removed?

File: src/app/weak-supervision/components/weak-supervision/weak-supervision.component.html [hunk 4]
```diff
@@ -548,17 +549,109 @@ <h1 class="text-lg text-gray-900 mb-2 text-center mb-4">Add new active learning<
...
+                                                        viewBox="0 0 24 24" stroke-width="2" stroke="currentColor"
+                                                        fill="none" stroke-linecap="round" stroke-linejoin="round">
+                                                        <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
+                                                        <circle cx="9" cy="7" r="4"></circle>
+                                                        <path d="M3 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"></path>
+                                                        <path d="M16 3.13a4 4 0 0 1 0 7.75"></path>
+                                                        <path d="M21 21v-2a4 4 0 0 0 -3 -3.85"></path>
+                                                    </svg>
+                                                </ng-template>
+                                            </ng-container> -->
```
Comment (FelixKirsch) [inline src/app/weak-supervision/components/weak-supervision/weak-supervision.component.html:650]: some comments left from developing

File: src/app/weak-supervision/components/weak-supervision/weak-supervision.component.html [hunk 5]
```diff
@@ -548,17 +549,109 @@ <h1 class="text-lg text-gray-900 mb-2 text-center mb-4">Add new active learning<
...
                                         <div class="text-gray-900 float-left">
                                             {{source.name}}
                                         </div>
+                                        <!-- </div> -->
```
Comment (FelixKirsch) [inline src/app/weak-supervision/components/weak-supervision/weak-supervision.component.html:654]: see last comment

**Review by FelixKirsch — APPROVED** [review]
Looks good :)

File: src/app/util/route-manager.ts (hunk 3 above)
Comment (JWittmeyer) [reply src/app/util/route-manager.ts:70]: removed

File: src/app/projects/components/project-settings/helper/label-helper.ts (hunk 2 above)
Comment (JWittmeyer) [reply src/app/projects/components/project-settings/helper/label-helper.ts:146]: changed

File: src/app/weak-supervision/components/weak-supervision/weak-supervision.component.html (hunk 4 above)
Comment (JWittmeyer) [reply src/app/weak-supervision/components/weak-supervision/weak-supervision.component.html:650]: removed

**Pull request merged**